kafka-rest/src/main/java/io/confluent/kafkarest/ConsumerManager.java

```diff
@@ -354,6 +358,58 @@ private ConsumerState getConsumerInstance(String group, String instance) {
     ConsumerConnector createConsumer(ConsumerConfig config);
   }
 
+  class ReadFutureTask<V> extends FutureTask<V> {
+
+    private Runnable myTask;
```

Comment:
Couldn't we use the more specific type `RunnableReadTask` and avoid the cast below?

Also a couple nits: 

1) perhaps a better name is simply `readTask`?
2) could this field be final?

Reply:
good idea,thanks

Reply:
@huangyiminghappy I think we needed to rebase. I went ahead and did it since you have been waiting so patiently to get this patch in.